### PR TITLE
Added connecting api for ruby-binlog closed? api

### PR DESCRIPTION
**Pull request by mmasashi**

### CODE REVIEW

File: include/binlog_api.h [hunk 1]
```diff
@@ -114,6 +114,10 @@ class Dummy_driver : public system::Binary_log_driver
   {
     return ERR_OK;
   }
+  virtual my_bool connecting()
```
Comment (makinada) [inline include/binlog_api.h]: `my_bool` is MySQL proprietary.  Better not to expose it outside of mysql-replication-listener.  ruby-binlog includes this header file but it doesn't use `my_bool` currently.  Instead, you can use either `bool` or `int` depending on how mysql-replication-listener handles boolean values in its external API.

File: src/file_driver.cpp [hunk 2]
```diff
@@ -70,6 +70,7 @@ using namespace std;
     {
       return ERR_FAIL;
     }
+    m_connecting= 1;
```
Comment (makinada) [inline src/file_driver.cpp]: Looking at how it's set/unset, `m_connected` is probably a better name.

**Comment by mmasashi** [comment]
Thanks for the review!  I agree with both of your comments.

File: include/binlog_api.h [hunk 3]
```diff
@@ -114,6 +114,10 @@ class Dummy_driver : public system::Binary_log_driver
   {
     return ERR_OK;
   }
+  virtual my_bool connected()
```
Comment (makinada) [inline include/binlog_api.h]: This `my_bool` should also be changed to `bool`

**Pull request merged**